src/js/views/maps/viewfinder/PredictionView.js

```diff
@@ -0,0 +1,114 @@
+     * for the user when they are searching for a place on a map.
+     * @classcategory Views/Maps
+     * @name PredictionView
+     * @extends Backbone.View
+     * @screenshot views/maps/viewfinder/PredictionView.png
+     * @since x.x.x
+     * @constructs PredictionView
+     */
+    const PredictionView = Backbone.View.extend({
+      /**
+       * The type of View this is
+       * @type {string}
+       */
+      type: 'PredictionView',
+
+      /**
+       * The HTML class to use for this view's outermost element.
+       * @type {string}
+       */
+      className: BASE_CLASS,
+
+      /**
+       * The HTML element to use for this view's outermost element.
+       * @type {string}
+       */
+      tagName: 'li',
+
+      /**
+       * The HTML classes to use for this view's HTML elements.
+       * @type {Object<string,string>}
+       */
+      classNames: {
+        content: `${BASE_CLASS}__content`,
+      },
+
+      /**
+      * The events this view will listen to and the associated function to call.
+      * @type {Object}
+      */
+      events: { click: 'select' },
+
+      /** 
+       * Values meant to be used by the rendered HTML template.
+       */
+      templateVars: {
+        classNames: {},
+        isFocused: false,
+      },
+
+      /**
+       * @typedef {Object} PredictionViewOptions
+       * @property {Prediction} The Prediction model associated with this
+       * autocompletion prediction.
+       * @property {ViewfinderModel} The model associated with the parent view.
+       * @property {number} The position of this prediction within the parent's
+       * full list of predictions.
+       */
+      initialize({ index, predictionModel, viewfinderModel }) {
+        this.predictionModel = predictionModel;
+        this.viewfinderModel = viewfinderModel;
+        this.index = index;
+
+        this.templateVars = {
+          ...this.templateVars,
+          classNames: this.classNames,
+          description: this.predictionModel.get('description'),
+        };
+
+        this.setupListeners();
+      },
+
+      /**
+       * Setup all event listeners on ViewfinderModel.
+       */
+      setupListeners() {
+        this.listenTo(this.viewfinderModel, 'change:focusIndex', () => {
+          this.render();
+        });
+      },
+
+      /**
+       * Event handler function that selects this element, deselecting any other 
+       * sibling list elements.
+       */
+      select(event) {
+        event.stopPropagation();
+        this.viewfinderModel.selectPrediction(this.predictionModel);
+      },
+
+      /**
+       * Render the view by updating the HTML of the element.
+       * The new HTML is computed from an HTML template that
+       * is passed an object with relevant view state.
+       * */
+      render() {
+        const focusIndex = this.viewfinderModel.get('focusIndex');
+        this.templateVars.isFocused = focusIndex !== -1 && focusIndex === this.index;
```

Comment:
nit: the -1 check is probably not necessary since this.index won't be -1?

Reply:
Done